src/main/java/com/iexec/sms/secret/web2/Web2Secrets.java

```diff
@@ -43,13 +42,72 @@ public class Web2Secrets {
     private List<Secret> secrets;
 
     Web2Secrets(String ownerAddress) {
-        this.ownerAddress = ownerAddress;
-        this.secrets = new ArrayList<>();
+        this(ownerAddress, List.of());
+    }
+
+    public Web2Secrets(String ownerAddress, List<Secret> secrets) {
+        this.ownerAddress = ownerAddress.toLowerCase();
+        this.secrets = Collections.unmodifiableList(secrets);
     }
 
     public Optional<Secret> getSecret(String secretAddress) {
         return secrets.stream()
                 .filter(secret -> secret.getAddress().equals(secretAddress))
                 .findFirst();
     }
+
+    /**
+     * Copies the current {@link Web2Secrets} object, while adding given {@link Secret} to the secrets list.
+     *
+     * @param secretAddress    Address of the new secret.
+     * @param secretValue      Value of the new secret.
+     * @param isEncryptedValue Whether this value is encrypted.
+     * @return A new {@link Web2Secrets} instance, with a new {@link Secret} element.
+     * @throws SecretAlreadyExistsException thrown when a secret with same address already exists.
+     */
+    public Web2Secrets addNewSecret(String secretAddress, String secretValue, boolean isEncryptedValue)
```

Comment:
When do we need to `addNewSecret` with encrypted value?

Reply:
Always? I mean, we don't ever add any deciphered secret. We'll probably remove this tweaked feature in our next PR.

Reply:
Done in https://github.com/iExecBlockchainComputing/iexec-sms/pull/134/commits/d74b75d0af7bfa4e1f34e1fbe4ed8cb46925fc3f :)

Reply:
Thank you :pray: